File: _01_Languages/Kotlin/src/_03_clases/_05_Polimorfismo.kt
/*
 * 🧩 Enunciado:
 * Crea una lista de empleados y gerentes.
 * Usa polimorfismo para llamar a mostrarDatos() dinámicamente.
 */

fun main() {
    val empleados: List<Empleado> = listOf(
        Empleado("Carlos", 1200.0),
        Gerente("Laura", 1800.0, 500.0),
        Empleado("Pablo", 1000.0),
        Gerente("Marta", 2000.0, 700.0)
    )

    for (emp in empleados) {
        emp.mostrarDatos() // Polimorfismo
    }
}

File: _01_Languages/CSharp/src/_02_poo/_05_Polimorfismo.cs
/*
 * 🧩 Enunciado:
 * Crea una lista de empleados donde algunos sean gerentes y otros empleados.
 * Usa polimorfismo para llamar a `MostrarDatos()` de forma dinámica.
 */

using System;
using System.Collections.Generic;

class _05_Polimorfismo
{
    static void Main()
    {
        List<Empleado> empleados = new List<Empleado>()
        {
            new Empleado("Carlos", 1200),
            new Gerente("Laura", 1800, 500),
            new Empleado("Pablo", 1000),
            new Gerente("Marta", 2000, 700)
        };

        foreach (Empleado emp in empleados)
        {
            emp.MostrarDatos(); // Polimorfismo en acción
        }
    }
}
